lgn-messages/src/types/v1/query/tasks.rs

```diff
@@ -31,148 +31,60 @@ pub struct QueryInput
 #[derive(Debug, Clone, Deserialize, Serialize)]
 pub enum QueryStep
 {
+    /// Combine the rows and revelation proving for tabular queries in one task,
+    /// next step is Groth16
     #[serde(rename = "1")]
-    Prepare(Vec<QueryInputPart>),
+    Tabular(
+        Vec<MatchingRowInput>,
+        RevelationInput,
+    ),
 
+    /// Aggregation batching queries, next step is Revelation
     #[serde(rename = "2")]
-    Revelation(RevelationInput),
-}
-
-#[derive(Dbg, Clone, PartialEq, Deserialize, Serialize)]
-pub struct QueryInputPart
-{
-    pub proof_key: ProofKey,
-
-    pub embedded_proof_input: Option<EmbeddedProofInputType>,
+    Aggregation(AggregationInput),
 
-    pub aggregation_input_kind: Option<ProofInputKind>,
-}
-
-#[derive(Clone, PartialEq, Dbg, Deserialize, Serialize)]
-pub enum ProofInputKind
-{
-    /// Match in the end of path or not matched branch
-    #[serde(rename = "1")]
-    SinglePathLeaf(SinglePathLeafInput),
-
-    /// Match in the middle of path
-    #[serde(rename = "2")]
-    SinglePathBranch(SinglePathBranchInput),
-
-    /// Node in tree with only one child
+    /// Revelation step, we only handle aggregation revelation for now, next step is Groth16
     #[serde(rename = "3")]
-    PartialNode(PartialNodeInput),
-
-    /// Node in tree with both children
-    #[serde(rename = "4")]
-    FullNode(FullNodeInput),
-
-    NonExistence(NonExistenceInput),
-}
-
-#[derive(Clone, PartialEq, Dbg, Deserialize, Serialize)]
-pub struct FullNodeInput
-{
-    pub is_rows_tree_node: bool,
-
-    pub left_child_proof_location: ProofKey,
-
-    #[dbg(placeholder = "...")]
-    pub left_child_proof: Vec<u8>,
-
-    pub right_child_proof_location: ProofKey,
-
-    #[dbg(placeholder = "...")]
-    pub right_child_proof: Vec<u8>,
-}
-
-#[derive(Clone, PartialEq, Dbg, Deserialize, Serialize)]
-pub struct PartialNodeInput
-{
-    pub proven_child_position: ChildPosition,
-
-    pub proven_child_proof_location: ProofKey,
-
-    #[dbg(placeholder = "...")]
-    pub proven_child_proof: Vec<u8>,
-
-    pub unproven_child_info: Option<NodeInfo>,
-
-    pub is_rows_tree_node: bool,
-}
-
-#[derive(Clone, PartialEq, Dbg, Deserialize, Serialize)]
-pub enum EmbeddedProofInputType
-{
-    #[serde(rename = "1")]
-    RowsTree(RowsEmbeddedProofInput),
-
-    #[serde(rename = "2")]
-    IndexTree(IndexEmbeddedProofInput),
+    Revelation(RevelationInput),
 }
 
-#[derive(Dbg, Clone, PartialEq, Deserialize, Serialize)]
-pub struct RowsEmbeddedProofInput
+#[derive(Clone, Dbg, PartialEq, Deserialize, Serialize)]
+pub struct MatchingRowInput
```

Comment:
Can you make some comments in all the new structs and fields you introduce please ? Introduce the notion of aggregation and chunks and matching row etc.
Not long text but something initial to guide the reader and point to the MP2 repo in the comments to read further.